plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1453,9 +1468,30 @@ private void finishWrite(ConnectorSession session, IcebergTableHandle table, Col
             cleanExtraOutputFiles(session, writtenFiles.build());
         }
 
+        if (!fullyDeletedFiles.isEmpty()) {
+            try {
+                FileSystem fileSystem = hdfsEnvironment.getFileSystem(new HdfsContext(session), new Path(table.getTableLocation()));
+                for (List<CommitTaskData> commitTasksToCleanUp : fullyDeletedFiles.values()) {
+                    for (CommitTaskData commitTaskData : commitTasksToCleanUp) {
+                        if (!fileSystem.delete(new Path(commitTaskData.getPath()), false)) {
+                            log.warn("Failed to clean up uncommitted position delete file: %s", commitTaskData.getPath());
+                        }
+                    }
+                }
+            }
+            catch (IOException e) {
+                log.warn(e, "Failed to clean up uncommitted position delete files");
+            }
+        }
+
         rowDelta.validateDataFilesExist(referencedDataFiles.build());
         try {
             rowDelta.commit();
+            if (!fullyDeletedFiles.isEmpty()) {
```

Comment:
`RowDelta` is unconditionally constructed and committed, even when all files wholly removed.
Why not construct and commit `DeleteFiles` unconditionally? Is there any cost associated with that?

Reply:
Looks like the both write a new ManifestList even if there are no rows changed so they both should be conditional.

Reply:
add code comments too